src/commands/talk.ts

```diff
@@ -0,0 +1,224 @@
+// External dependencies
+import { writeFileSync, createReadStream } from 'node:fs';
+import { isNumber } from 'lodash';
+import {
+	SlashCommandBuilder,
+	AttachmentPayload,
+	ChannelType,
+	PermissionFlagsBits,
+	CommandInteractionOptionResolver,
+} from 'discord.js';
+import {
+	createAudioResource,
+	createAudioPlayer,
+	joinVoiceChannel,
+	getVoiceConnection,
+	AudioPlayerStatus,
+	VoiceConnectionStatus,
+	NoSubscriberBehavior,
+	entersState,
+	AudioResource,
+} from '@discordjs/voice';
+import { say, Speaker } from 'dectalk';
+import { fileSync } from 'tmp';
+
+// Internal depedencies
+import * as logger from '../logger';
+
+const builder = new SlashCommandBuilder()
+	.setName('talk')
+	.setDescription('Uses Dectalk to speak the given message')
+	.addStringOption(option =>
+		option.setRequired(true).setName('message').setDescription('The message to speak')
+	)
+	.addIntegerOption(option => {
+		option.setRequired(false).setName('speaker').setDescription('Whose voice to use');
+
+		Object.values(Speaker)
+			.filter(isNumber)
+			.forEach(value => {
+				option = option.addChoices({ name: Speaker[value] as string, value: value });
```

Comment:
Better to handle the `undefined` case (using `??` or literally anything else) than to use `as` to explicitly ignore it.